invenio_records_resources/records/api.py

```diff
@@ -66,13 +66,14 @@ def get_by_key(cls, record_id, key):
     @classmethod
     def list_by_record(cls, record_id, with_deleted=False):
         """List all record files by record ID."""
-        query = cls.model_cls.query.filter(cls.model_cls.record_id == record_id)
+        with db.session.no_autoflush:
```

Comment:
I am not sure about this change, could you explain how does it help?

Reply:
I think it was leftover from @jrcastro2  but in principle this is the fix we apply in other places instructing sqlalchemy not to trigger UPDATE queries when fetching from db. Is that the question?

Reply:
It helps very little, concretely it prevents 2 more queries to the DB every time this is called, the `no_autoflush` is useful when we are only querying, without executing any modifications on the data. Otherwise it can lead to inconsistent data. We can remove it if it's too complicated as the real gains are very little in this case.